### ISSUE TITLE
FE: UX/UI improvements Part 2

### ISSUE THREAD

**Issue by SzymonSt** [SzymonSt/docker-signalone#108]
## Contrast Ratio (minimum contrast ratio should be 4,5:1)

- [x] Text input form element - placeholder
Issue: Currently, the placeholder color of the text input field in the form has a contrast ratio of 1,77. The minimum contrast ratio should be 4,5. 
Suggestion:  Maybe it would be better to darken the border of that input a little bit.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/b2293279-3de7-4ec9-a2a9-d692080dd477)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/c239bb35-76c2-457a-bd4b-726a002799de)

- [ ] Date picker
Issue: Currently, a lot of date picker elements do not meet minimum contrast ratio criteria. 

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/1ecbb123-5786-4e08-8bf6-6da3a2633167)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/3514ae75-0fb8-4a0f-869d-e96e4082bbf2)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/45a54113-2673-4cd5-a96c-8ef8007e3635)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/3eb0015d-e963-4901-b675-25cef29c0ff6)

- [ ] Toast message - success
Issue: Currently, the success toast message does not meet the minimum contrast ratio criteria. 

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/008aa068-e830-4de2-9e40-4e0089b06555)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/8a117a07-6c65-429e-a80f-26e3f9267d94)

## Keyboard interaction/styling

- [ ] Tabbing - the user should be able to `tab` through the entire app
Issue: Currently, the ability to navigate in the app with a keyboard only is not possible. 
Users should be able to tab through the language choice, issues list, and interact with all form fields with a keyboard only.

https://github.com/SzymonSt/docker-signalone/assets/68923217/b2176005-ee95-41b5-a680-ef6deedf2924

- [ ] List controls - unintended scroll after clicking the "space" key 
Issue: Currently, if the list control element is focused and the user hits `space` then the page scrolls which is unintuitive.
Suggestion: execute `event.preventDefault()` if the user hits `space` key while focusing the control button

https://github.com/SzymonSt/docker-signalone/assets/68923217/489a0e25-55b2-4a3e-a9ae-0bab2253f315

- [x] Focus indicator styling - settings icon
Issue: When the user focuses the settings icon with the keyboard the focus indicator (the border around the icon) looks weird.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/c4ca4534-628e-4e53-8b17-91722fb3e090)

- [ ] Tabbing - disabled controls shouldn't get focus
Issue: Currently, the user can focus the disabled list controls with the keyboard
Suggestion: The disabled list controls buttons should have `tab-index="-1"`  applied, also if it's a button it should have a `disabled`  and `aria-disabled="true"` attributes

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/50e80dbd-0423-4b8d-9610-cdb00efc2764)

- [ ] Keyboard focus styling - search button

Issue: If the user focuses the `search` button with the keyboard it changes the background color to blue which probably does not fit our app color palette 
Suggestion: Provide styling for the `:focus-visible` pseudo-class

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/535e96ca-d10c-4eae-9d44-220f5b5c62ae)

- [ ] Keyboard Interaction - dropdown

Issue: Currently user can't use the dropdown field with a keyboard only. 
Suggestion: We should provide proper styling indicating which option is already focused and give the user the ability to navigate with the `arrow` keys and select the desired option with the `enter` key.

- [ ] Keyboard Interaction - date field
Issue: Currently user can't open the date picker with a keyboard only.
Suggestion: When the date field gets focused via the `Tab` key, we should display the date picker popup to give the user the ability to choose the desired date.

## Visual improvements 

- [x] Issue detailed view - scroll overflows the container

(maybe it's hard to see, but the container has rounded corners and the scrollbar overflows the container which doesn't look good)

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/7a50de82-aadf-4bb3-a324-25fcc9146828)

- [ ] Date Picker popup - unify the styling 
Issue: Currently, almost all elements in our app have rounded corners. Probably the only exception is the date picker popup which has content with a rounded container but doesn't have rounded corners on the external container.
Suggestion: all elements should follow the same styling convention i.e. rounded corners.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/7169b1ba-98d0-4b21-a118-b069cce96a96)

- [ ] Dropdown - flashing/flickering effect
Issue: When the user focuses the dropdown by clicking many times in the dropdown "control" element when the flashing effect is happening, which doesn't look good.

https://github.com/SzymonSt/docker-signalone/assets/68923217/171c68f5-c09b-4f32-9cbb-7252bd39bcb9

- [ ] Dropdown - control-menulist parts visual glitch
Issue: After opening the dropdown menu the menu list is slightly moved (by 1px) from the dropdown control part. It is hard to see on the screenshot, but you can see it locally in the selected area.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/a5339137-4569-4f38-9fba-46669ef33be9)

- [ ] Dropdown - white space near menu list corners
Issue: Minor issue, but there is a whitespace near the dropdown menu list corners. 
Suggestion: unify the `border-radius` property for the menu list and menu item

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/294001c9-c644-44c4-9ed8-956ce6e41f9a)

- [x] Checkbox - align the label vertically
Issue: the checkbox element and its label are not aligned vertically 

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/b13992d0-45d5-4d88-a05a-c157409b3f9c)

- [ ] Form - expand arrows - unify the colors
Issue: The form should consist of similar elements. Currently, the following expand arrows do not have the same color, we should unify this.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/b05e6127-70e2-4216-9f0c-5cbcb518af5c)

- [ ] Form header - remove the focus indicator after clicking with the mouse
Issue: Currently we show the blue focus indicator after clicking on the `Search criteria`. We should display it while navigating via the keyboard, not with the mouse.

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/6adc83c2-cb6a-4023-bcde-298dce35e84b)

- [x] Form header - white space under the "header" after expanding the form
Issue: After expanding the `search criteria` the white rectangular appears under the whole header. 
Suggestion: it can be caused by the margin which is applied after expanding the form

(it's hard to see on the screenshot, but this white rectangular is the same size as the shown margin)

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/45c0399b-64dc-4e21-bd07-aad8df4c3a3f)
![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/90706754-81dc-40ee-a2eb-0263516d95be)

- [ ] Flag icons - border
Issue: Currently, country icons don't have any border which causes the flags to be hard to distinguish with the app background
Suggestion: provide the styling for focusing the icons with a keyboard (focus indicator)

![Image](https://github.com/SzymonSt/docker-signalone/assets/68923217/270130d8-7826-49cd-b1b3-412569ddb0fe)

- [x] Sidebar arrow is not centered in the docker extension 

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/c13c29d8-df2e-4070-b309-c1aca9ddb117)

- [x] Sidebar arrow is covered by the `Search Criteria` 

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/59e2faf5-a235-41b7-ae31-97fd666e592a)

- [x] Increase the width of the sidebar to make date fields fully visible

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/eadf7bb7-696a-4648-bf97-d8fadcf81770)

- [x] Checkbox should have the same focus ring color as other inputs 

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/840e2be0-15c6-4707-8ba3-995245d158b7)

- [x] The message about not issue selected should be divided into 2 lines and centered

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/f283402d-5179-4788-92f9-4affff87611e)

- [x] Adjust activate agent toast message

Currently, the message says: "Selected Agent activated"
We should change it to: "Agent activated"

- [x] The message for activating and deactivating the agent is the same

We should display "Agent deactivated" message if user turns off the agent

![image](https://github.com/SzymonSt/docker-signalone/assets/68923217/e2b90400-7169-4712-a606-d2a1ddedbad4)
